python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -626,6 +653,43 @@ def _create_initial_sensor_cursors_from_raw_cursor(
 
         return result
 
+    def _rename_auto_materialize_states(
+        self,
+        instance: DagsterInstance,
+        all_auto_materialize_states: Mapping[str, InstigatorState],
+        sensors_and_repos: Sequence[Tuple[ExternalSensor, ExternalRepository]],
+    ) -> Mapping[str, InstigatorState]:
+        """Searches for sensors named `default_auto_materialize_sensor` and migrates their state
+        to a sensor in the same repository named `default_automation_condition_sensor`.
+        """
+        result = dict(all_auto_materialize_states)
+
+        for instigator_state in all_auto_materialize_states.values():
+            instigator_state_name = instigator_state.origin.instigator_name
+            instigator_state_repository_origin = instigator_state.origin.repository_origin
+            # only migrate instigators with the name "default_auto_materialize_sensor"
+            if instigator_state_name != "default_auto_materialize_sensor":
+                continue
+            for sensor, repo in sensors_and_repos:
+                # find the corresponding sensor with the updated name
+                if (
+                    sensor.name == "default_automation_condition_sensor"
+                    and repo.get_external_origin() == instigator_state_repository_origin
+                ):
+                    new_auto_materialize_state = InstigatorState(
+                        sensor.get_external_origin(),
```

Comment:
i am a little unsure why we need sensors_and_repos here at all - can the new origin to use not always be derived from the old origin?

i.e. could this not be

new_sensor_origin = instigator_state.origin._replace(instigator_name="default_automation_condition_sensor")

and then we create it, whether or not there is a sensor with that name?

Not sure if it matters in practice but seems a bit simpler

Reply:
Yeah the main thing here was just having access to the selector id, which given your comment below is actually possible to get without the sensor itself. This seems a lot better, and guards against a case where the given code location is unavailable at the time of the migration